snuba/processor.py

```diff
@@ -103,19 +106,22 @@ def _unicodify(s):
     return str(s).encode("utf8", errors="backslashreplace").decode("utf8")
 
 
-def _hashify(h):
+def _hashify(h: str) -> str:
     if HASH_RE.match(h):
         return h
     return md5(force_bytes(h)).hexdigest()
 
 
-def _ensure_valid_date(dt):
+epoch = datetime(1970, 1, 1)
+
+
+def _ensure_valid_date(dt: Optional[datetime]) -> Optional[datetime]:
     if dt is None:
         return None
-    seconds = (dt - datetime(1970, 1, 1)).total_seconds()
+    seconds = (dt - epoch).total_seconds()
     if _collapse_uint32(seconds) is None:
         return None
-    return dt
+    return epoch + timedelta(seconds=seconds)
```

Comment:
This is kind of silly, but ensures the value actually does fit within the range, since it might be truncated by `_collapse_uint32` since `total_seconds` returns a `float`.